### ISSUE TITLE
Installation issue: netcdf-cxx4

### ISSUE THREAD

**Issue by payerle** [LLNL/spack#16725]
```console
$ spack install <spec>
...
```

### Additional information

* [spack-build-out.txt]()
* [spack-build-env.txt]()

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by payerle** [comment]
@WardF
* **Spack:** 0.14.2-1112-445cae5c8
[spack-build-env.txt](https://github.com/spack/spack/files/4652590/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/4652591/spack-build-out.txt)
[spack-configure-args.txt](https://github.com/spack/spack/files/4652592/spack-configure-args.txt)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-x86_64

**Comment by payerle** [comment]
Getting a failure on netcdf-cxx4 due to illegal configure args.

E.g., 
spack install netcdf-cxx4@4.3.1%gcc@8.4.0 +dap +doxygen +jna +ncgen4 +netcdf4 +pic ~pnetcdf +shared +static arch=linux-rhel8-x86_64

Dies with error
    28     ==> 29125: netcdf-cxx4: Executing phase: 'configure'
     29     ==> [2020-05-19-11:44:18.476779, 29134] '/tmp/payerle/spack-stage/spack-stage-netcdf-c
            xx4-4.3.1-ubzluaakvqaghjnlpdpr4cz4oy4kv5su/spack-src/configure' 'CFLAGS=-fPIC' 'CPPFLA
            GS=-I/software/spack-software/2020.05.14/linux-rhel8-x86_64/gcc-8.4.0/netcdf-c-4.7.3-l
            lhykha5mn7j5lg5fhrepujjgvfsei3j/include' '--prefix=/software/spack-software/2020.05.14
            /linux-rhel8-x86_64/gcc-8.4.0/netcdf-cxx4-4.3.1-ubzluaakvqaghjnlpdpr4cz4oy4kv5su' '--enable-static' '--enable-shared' '--with-pic' '--enable-dap' '--enable-jna' '--disable-pnetcdf' '--enable-netcdf-4' '--enable-ncgen4' '--enable-doxygen'
  >> 30     configure: WARNING: unrecognized options: --enable-dap, --enable-jna, --disable-pnetcdf, --enable-netcdf-4, --enable-ncgen4

Basically, it looks like #16047/commit  993491c added new variants and corresponding config args for netcdf-cxx4, but the added configure args are not supported by the netcdf-cxx4 configure script (I checked both 4.3.0 and 4.3.1; --enable-jna, et al. do not appear in configure --help for either version)

**Comment by WardF** [comment]
These arguments appear to be for the netcdf-c library configuration script.
